### ISSUE TITLE
How to use party.js in react and confetti should emit infinite time?

### ISSUE THREAD

**Issue by Gyan555** [yiliansource/party-js#85]
Compiled with warnings.

<details>
<summary>Warnings:</summary>

```
Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\circle.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\circle.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\color.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\color.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\gradient.ts' file: Error: ENOENT: no such file or directory, 
open 'E:\abc\node_modules\party-js\src\components\gradient.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\index.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\index.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\numericSpline.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\numericSpline.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\rect.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\rect.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\spline.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\spline.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\vector.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\vector.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\containers.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\containers.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\debug.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\debug.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\index.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\index.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\emitter.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\particles\emitter.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\options\emissionOptions.ts' file: Error: ENOENT: no such file 
or directory, open 'E:\abc\node_modules\party-js\src\particles\options\emissionOptions.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\options\emitterOptions.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\particles\options\emitterOptions.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\options\index.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\particles\options\index.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\options\renderOptions.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\particles\options\renderOptions.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\particle.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\particles\particle.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\renderer.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\particles\renderer.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\scene.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\scene.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\settings.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\settings.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\systems\math.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\systems\math.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\systems\modules.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\systems\modules.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\systems\random.ts' file: Error: ENOENT: no such file or directory, open 
'E:\abc\node_modules\party-js\src\systems\random.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\systems\shapes.ts' file: Error: ENOENT: no such file or directory, open 
'E:\abc\node_modules\party-js\src\systems\shapes.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\systems\sources.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\systems\sources.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\systems\variation.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\systems\variation.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\templates\confetti.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\templates\confetti.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\templates\index.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\templates\index.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\templates\sparkles.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\templates\sparkles.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\util\config.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\util\config.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\util\index.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\util\index.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\util\lazy.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\util\lazy.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\util\rotation.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\util\rotation.ts'

Failed to parse source map from 'E:\abc\node_modules\party-js\src\util\rules.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\util\rules.ts'

Search for the keywords to learn more about each warning.
To ignore, add // eslint-disable-next-line to the line before.

WARNING in ./node_modules/party-js/lib/components/circle.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\circle.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\circle.ts'

WARNING in ./node_modules/party-js/lib/components/color.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\color.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\color.ts'

WARNING in ./node_modules/party-js/lib/components/gradient.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\gradient.ts' file: Error: ENOENT: no such file or directory, 
open 'E:\abc\node_modules\party-js\src\components\gradient.ts'

WARNING in ./node_modules/party-js/lib/components/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\index.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\index.ts'

WARNING in ./node_modules/party-js/lib/components/numericSpline.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\numericSpline.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\numericSpline.ts'

WARNING in ./node_modules/party-js/lib/components/rect.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\rect.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\rect.ts'

WARNING in ./node_modules/party-js/lib/components/spline.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\spline.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\spline.ts'

WARNING in ./node_modules/party-js/lib/components/vector.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\components\vector.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\components\vector.ts'

WARNING in ./node_modules/party-js/lib/containers.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\containers.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\containers.ts'

WARNING in ./node_modules/party-js/lib/debug.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\debug.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\debug.ts'

WARNING in ./node_modules/party-js/lib/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\index.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\index.ts'

WARNING in ./node_modules/party-js/lib/particles/emitter.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\emitter.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\particles\emitter.ts'

WARNING in ./node_modules/party-js/lib/particles/options/emissionOptions.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\options\emissionOptions.ts' file: Error: ENOENT: no such file 
or directory, open 'E:\abc\node_modules\party-js\src\particles\options\emissionOptions.ts'

WARNING in ./node_modules/party-js/lib/particles/options/emitterOptions.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\options\emitterOptions.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\particles\options\emitterOptions.ts'

WARNING in ./node_modules/party-js/lib/particles/options/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\options\index.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\particles\options\index.ts'

WARNING in ./node_modules/party-js/lib/particles/options/renderOptions.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\options\renderOptions.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\particles\options\renderOptions.ts'

WARNING in ./node_modules/party-js/lib/particles/particle.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\particle.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\particles\particle.ts'

WARNING in ./node_modules/party-js/lib/particles/renderer.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\particles\renderer.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\particles\renderer.ts'

WARNING in ./node_modules/party-js/lib/scene.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\scene.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\scene.ts'

WARNING in ./node_modules/party-js/lib/settings.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\settings.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\settings.ts'

WARNING in ./node_modules/party-js/lib/systems/math.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\systems\math.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\systems\math.ts'

WARNING in ./node_modules/party-js/lib/systems/modules.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\systems\modules.ts' file: Error: ENOENT: no such file or directory, open 'E:\anc\node_modules\party-js\src\systems\modules.ts'

WARNING in ./node_modules/party-js/lib/systems/random.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\systems\random.ts' file: Error: ENOENT: no such file or directory, open 
'E:\abc\node_modules\party-js\src\systems\random.ts'

WARNING in ./node_modules/party-js/lib/systems/shapes.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\systems\shapes.ts' file: Error: ENOENT: no such file or directory, open 
'E:\abc\node_modules\party-js\src\systems\shapes.ts'

WARNING in ./node_modules/party-js/lib/systems/sources.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\systems\sources.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\systems\sources.ts'

WARNING in ./node_modules/party-js/lib/systems/variation.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\systems\variation.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\systems\variation.ts'

WARNING in ./node_modules/party-js/lib/templates/confetti.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\templates\confetti.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\templates\confetti.ts'

WARNING in ./node_modules/party-js/lib/templates/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\templates\index.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\templates\index.ts'

WARNING in ./node_modules/party-js/lib/templates/sparkles.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\templates\sparkles.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\templates\sparkles.ts'

WARNING in ./node_modules/party-js/lib/util/config.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\util\config.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\util\config.ts'

WARNING in ./node_modules/party-js/lib/util/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\util\index.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\util\index.ts'

WARNING in ./node_modules/party-js/lib/util/lazy.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\util\lazy.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\util\lazy.ts'

WARNING in ./node_modules/party-js/lib/util/rotation.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\util\rotation.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\util\rotation.ts'

WARNING in ./node_modules/party-js/lib/util/rules.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'E:\abc\node_modules\party-js\src\util\rules.ts' file: Error: ENOENT: no such file or directory, open 'E:\abc\node_modules\party-js\src\util\rules.ts'
```

</details>

I had used npm intsall party-js
```
useEffect(() => {
    setInterval(() => {
      party.confetti(document.getElementById("button"), {
        count: party.variation.range(0, 5),
        size: party.variation.range(1.8, 2),
        spread: party.variation.range(15, 20),
      });
    }, 1000 / 60);
  }, []);
```

**Comment by tonwayer** [comment]
Your project looks based on TypeScript. The warnings display because you did not import types of party-js. But unfortunately, we don't have @types/party-js yet. It's time to make @types/party-js module!

**Comment by yiliansource** [comment]
@tonwayer The library does ship with type definitions, since it's written in TypeScript itself.

@Gyan555 Have you checked, are the files that are mentioned as missing actually missing? Seems like some issue with source maps.

**Comment by tonwayer** [comment]
No. It does not have src folder inside `node_modules\party-js`. I've just install this module using `npm install party-js`

**Comment by yiliansource** [comment]
I see, it does seem like the source maps are generated incorrectly, I'll look into it!

**Comment by yiliansource** [comment]
@tonwayer @Gyan555 I just released patch `2.1.3`, which removes source maps from the bundle. Let me know how this works out for you.

**Comment by yiliansource** [comment]
Closing due to inactivity.
